plugin-enclave-gradle/src/main/kotlin/com/r3/conclave/plugin/enclave/gradle/GradleEnclavePlugin.kt

```diff
@@ -427,7 +496,19 @@ class GradleEnclavePlugin @Inject constructor(private val layout: ProjectLayout)
                 task.doFirst(RenameSignedEnclaveJarTask())
             }
 
-            target.artifacts.add(typeLowerCase, signedEnclaveJarTask.get().archiveFile)
+            target.afterEvaluate {
+                val runtimeType = conclaveExtension.runtime.get()
+
+                when (runtimeType) {
+                    RuntimeType.Gramine -> {
+                        target.artifacts.add(typeLowerCase, buildGramineEnclaveJarTask.get().archiveFile)
+                    }
+
+                    RuntimeType.GraalVM -> {
+                        target.artifacts.add(typeLowerCase, signedEnclaveJarTask.get().archiveFile)
```

Comment:
rename `signedEnclaveJarTask` to incorporate graalvm

Reply:
Actually no. Can you instead update `signedEnclaveJarTask` so that it does either the graalvm path or the gramine path. Because this task is what links the enclave module to the host one, i'd rather not have two different types and have a single gradle task for it.

Reply:
Mmm, should we use "signed" for Gramine as well? I.e. `gramine-direct` does not require signing as far as I understand.
Maybe we need an extra "final" task?

Reply:
For now use this "signed" task. I think we need to rename it anyway at some point, but for now it's fine.

Reply:
Is there any reason for using one task only at the end actually? I.e. why not be explicit and clear, i.e. separating the graphs for Gramine and Graal? I am trying to implement it and I notice that the code would become a bit conditional and ugly (changing the task after evaluation), and the user would actually risk more confusion by seeing the same task both for Gramine or Graal cases.

Reply:
Because the output of the enclave build process be encapsulated by a single task regardless of what the enclave config is. It makes it simpler to reason about, for us as the Conclave platform devs. This task isn't really exposed to the end user so they won't be confused, but I also don't think they would be confused anyway. In fact, I think the opposite of having separate tasks would be more confusing.

Reply:
Ok. I am trying to do that, but it looks to me that we are by purpose forcing some Gradle mechanism. How can we make `dependsOn` conditional and also make some of its dependencies conditional at runtime nicely?
My understanding of the Gradle mechanism is that we have that configuration and then execution phase. It really seems to me as if we actually need to create other 3 configurations and create the graph for them at compile/configuration time, and then execute the configuration according to the runtime selection.
I.e. I see runtimeType very similar to enclaveMode in the context of running these tasks.

Instead we seem to want to create our own graph at runtime. Is this really how Gradle is meant to behave? Is there a convention around that?

It seems to me that if we remove the "constraint" of using the same name of the last task, but instead we add the runtime type to all the tasks (as we do for the buildType) , our life would be easier and the logic would be straightforward.

Reply:
I see that this would require a change to the build directory, etc. And it's kind of messy to apply this now, as I expect a lot of code to have issues with it. Not sure what to do, I am tempted just to rename the final Gramine task with a "signed" naming in it, to make it similar to the previous one (and maybe rename the Graal one with "GraalVM" in it.
Any suggestion? Or do you know how to make dependencies conditional?

Reply:
OK no worries, let's leave this as is for now. We can readdress this later.